src/server/replica.cc

```diff
@@ -450,25 +454,43 @@ error_code Replica::InitiateDflySync() {
   // Make sure we're in LOADING state.
   CHECK(service_.SwitchState(GlobalState::ACTIVE, GlobalState::LOADING) == GlobalState::LOADING);
 
-  // Flush dbs.
-  JournalExecutor{&service_}.FlushAll();
-
   // Start full sync flows.
   state_mask_.fetch_or(R_SYNCING);
+  bool partial_sync = false;
   {
+    // Going out of the way to avoid using std::vector<bool>...
+    auto is_full_sync = std::make_unique<bool[]>(num_df_flows_);
     auto partition = Partition(num_df_flows_);
     auto shard_cb = [&](unsigned index, auto*) {
       for (auto id : partition[index]) {
-        auto ec = shard_flows_[id]->StartFullSyncFlow(sync_block, &cntx_);
+        auto ec = shard_flows_[id]->StartSyncFlow(sync_block, &cntx_, last_journal_LSNs_,
+                                                  is_full_sync.get()[id]);
         if (ec)
           cntx_.ReportError(ec);
       }
     };
-
     // Lock to prevent the error handler from running instantly
     // while the flows are in a mixed state.
     lock_guard lk{flows_op_mu_};
     shard_set->pool()->AwaitFiberOnAll(std::move(shard_cb));
+
+    bool all = true;
+    bool any = false;
+    for (bool is_full : absl::Span<bool>(is_full_sync.get(), num_df_flows_)) {
+      if (is_full)
+        any = true;
+      else
+        all = false;
+    }
+    if (all) {
+      JournalExecutor{&service_}.FlushAll();
+    } else if (any) {
+      last_journal_LSNs_.clear();
+      cntx_.ReportError(std::make_error_code(errc::state_not_recoverable),
```

Comment:
Shouldn't we `FlushAll()` in this case as well?
Or do I misunderstand, and this state won't lead to full sync? Is this an abort? (I guessed not because we're not `return`ing here, should we?)
Also, maybe add a log so users know what's going on?

Reply:
The flow ends in `RETURN_ON_ERR(cntx_.GetError());` and then we reconnect with empty `last_journal_LSNs_` and do a full sync.

Reply:
True but it it so big 😭 

why don't

```c++
size_t flows_partial = std::count(is_full_sync, is_full_sync + num_flows)

// Can't sync flows selectively, return error
if (flows_partial > 0 && flow_partial < num_flows)
  return error;

sync_type = flows_partial > 0 ? "partial" : "full"

// Clear state if not partial sync
if (sync_type == "full")
   JournalExecutor{&service_}.FlushAll()
```